### DISCUSSION TITLE
Replace "Null Value" with "All" in Cascading Parameter

### DISCUSSION THREAD

**Discussion by tnation1424** [eclipse-birt/birt#1581] [Q&A]
What is the quickest way to replace "Null Value" with the word "All" in a Cascading Parameter.  The example below after selecting an Installation the Employees are listed and BIRT uses Null Value for All.  I'm fine with that, but the people who run the report have a hard time dealing with it.
![image](https://github.com/eclipse-birt/birt/assets/149100941/34b1d10e-a1de-43e0-80f2-8f38a8fbe3a9)

**Comment by speckyspooky** [comment]
Yes, the would be a way to replace it but it is not directly the standard way.
The reason is that the value is fixed on source code level without translation property file.

But you can change a specific file in 5 lines and you will get "All" but be aware this would be a global change also for your parameter "ComboBox".

The way is to change the file **"ComboBoxParameterFragment.jsp"**.
You can find it in the folder structure of BIRT here:
- My application server: C:\Program Files\Apache Software Foundation\Tomcat 9.0_Tomcat-9.0.76\webapps\
- Birt-path: birt\webcontent\birt\pages\parameter\ComboBoxParameterFragment.jsp

The following file must be comment out:
```String outputLabel = ParameterAccessor.htmlEncode(( label == null)?IBirtConstants.NULL_VALUE_DISPLAY:label);```

The following line must be added directly below:
```
String outputLabel = (( label == null)?IBirtConstants.NULL_VALUE_DISPLAY:label);
if ( outputLabel == "Null Value" ) {
	outputLabel = "All";
}
outputLabel = ParameterAccessor.htmlEncode(outputLabel);
```

With this sequence you replace the standard label "Value Null" with "All".

### Screen example
![2024-03-07 21_48_38-ws - Compare _org eclipse birt report viewer_birt_webcontent_birt_pages_paramete](https://github.com/eclipse-birt/birt/assets/41593722/13c501e6-ce3d-4241-ae78-2070c30be192)

**Comment by tnation1424** [comment]
This is fantastic! Thank you!  Since this is for a Combo Box parameter, is there anything equivalent for a List Box?

**Reply by speckyspooky** [reply]
In my test also standard ListBox with singel selection and mutiple selection shows the new value "All".
Or do you mean another parameter?

**Comment by tnation1424** [comment]
Nope, I'm good.  I only thought because the name of the file was ComboBoxParameter it was only for a Combo Box parameter. You saying that "All" shows in your testing for List Box parameters works then I'm good.   "All" doesn't appear in List Box parameters for me after changing ComboBoxParameterFragment.jsp.   Maybe because I changed the ComboBoxParameterFragment.jsp in my Report Designer Environment and not in Application Server as you mention in your instructions.

**Reply by speckyspooky** [reply]
No there is no further file. And of course, you have to change both sides the designer side and the server because it is part of the BIRT-viewer-component.

If you have a further topic post a screen of the parameter configuration and I can take a look into your screen and the sources.
